src/irmin-containers/containers.ml

```diff
@@ -0,0 +1,86 @@
+(*
+ * Copyright (c) 2020 KC Sivaramakrishnan <kc@kcsrk.info>
+ * Copyright (c) 2020 Anirudh Sunder Raj <anirudh6626@gmail.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+open Lwt.Infix
+
+module type S = sig
+  type t
+
+  type repo
+
+  type branch
+
+  type key
+
+  type contents
+
+  module Store : Irmin.S
+
+  val init : ?bare:bool -> root:string -> unit -> repo Lwt.t
```

Comment:
```suggestion
  val init : ?bare:bool -> root:string -> unit -> repo Lwt.t
```

This initialiser isn't going to work for all backends (many don't have a FS root), as evidenced by the need for the backend-specific `Irmin_git.config` in the implementation here. Better to have the user supply the appropriate config for the backend.